ulauncher/api/server/ExtensionDownloader.py

```diff
@@ -182,7 +182,20 @@ def unzip(filename: str, ext_path: str) -> None:
     temp_ext_path = mkdtemp(prefix='ulauncher_dl_')
 
     with ZipFile(filename) as zipfile:
-        zipfile.extractall(temp_ext_path)
+        for zipinfo in zipfile.infolist():
+            filepath = zipfile.extract(zipinfo, temp_ext_path)
+            # Check if file is symlink from its attributes or zipfile did extract it as a symlink
+            # see https://learning-python.com/cgi/showcode.py?name=ziptools/ziptools/ziptools/zipsymlinks.py
+            if (zipinfo.external_attr >> 28) != 0xA or os.path.islink(filepath):
```

Comment:
I see, this is due to the fact that to git, and thus inside this zip archive, the symlink is really just a text file with the path as contents. So this snippet is trying to convert that normal text file into a true symlink, as long as it is within the extension directory. Could you add a few comments calling out the fact that most/all of the times zips from Git will have the symbolic links represented as text files? That might help future us understand what is going on here and why.

Reply:
Actually it is not only to git but to any normal  zip file. The current unzip function cannot handle any of these due to `ZipFile` not handling symlinks. You can test that by creating locally a zip with symlinks inside and pass it to Ulauncher's unzip function (that is what I actually tested in the first place). The `zipinfo.external_attr >> 28` check is generic as can be seen [here](https://learning-python.com/cgi/showcode.py?name=ziptools/ziptools/ziptools/zipsymlinks.py) at the `isSymlink` function.

I tested my function against a directory downloaded with Ulauncher's download_zip function and also a normal zip created locally by me totally unrelated to git. It works in both cases.